clients/src/main/java/org/apache/kafka/common/record/DefaultRecordBatch.java

```diff
@@ -376,8 +378,8 @@ static void writeHeader(ByteBuffer buffer,
         buffer.putShort(position + PRODUCER_EPOCH_OFFSET, epoch);
         buffer.putInt(position + BASE_SEQUENCE_OFFSET, sequence);
         buffer.putInt(position + RECORDS_COUNT_OFFSET, numRecords);
-        long crc = Crc32.crc32(buffer, ATTRIBUTES_OFFSET, size - ATTRIBUTES_OFFSET);
-        buffer.putInt(position + CRC_OFFSET, (int) (crc & 0xffffffffL));
+        long crc = Crc32.crc32(buffer, ATTRIBUTES_OFFSET, sizeInBytes - ATTRIBUTES_OFFSET);
+        buffer.putInt(position + CRC_OFFSET, (int) crc);
```

Comment:
Seems fine. I probably copied this from `Utils.writeUnsignedInt`, but down-casting should be safe.